### PR TITLE
Update TestingControllerSample app

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #1950

@Rick-Anderson @scottaddie ...

* Upgraded the sample to 2.0
  * Removed Bower in favor of just providing in-sample static assets (or tapping CDNs for static assets via environment tag helper). Static assets, such as Bootstrap and Knockout, are required for the app to function. I don't want to force devs to use Bower for this.
  * Refactored the code that finds the target project to **NOT** rely upon the presence of a VS solution file. This makes the sample cross-IDE friendly.
* The topic could use a touch of work, but let me circle around on that later. I'm *on a mission* with the DNX TFM removals, and I have more projects to update/upgrade to get #1950 done.

btw -- Note that because *wwwroot* was added to the *.gitignore* file that I'm having to work around getting *wwwroot* assets updated for samples. Was wondering why that was done (or can it be removed)?

### CODE REVIEW

**Comment by scottaddie** [comment]
@guardrex I'm in favor of removing *wwwroot* from the *.gitignore* file. It's caused me nothing but pain with sample app updates. We'll need @Rick-Anderson to sign off on this before proceeding.

**Comment by Rick-Anderson** [comment]
On the rare occasions you need it use https://stackoverflow.com/questions/8006393/force-add-despite-the-gitignore-file

Without it we'll have too many files checked in when not needed.

**Comment by Rick-Anderson** [comment]
Will the force work or is that too much trouble on a folder?

**Comment by guardrex** [comment]
> too many files checked in

But *wwwroot* ... I thought it was only part of the samples in the repo, and the samples all need to have them, no?

The prob with me thus far is that part of my commit process uses GH Desktop (it's faster to click than type), so I end up having to do what a commenter on that SO states ...

> Another way of achieving it would be to temporary edit the gitignore file, add the file and then revert back the gitignore. A bit hacky i feel

**Comment by scottaddie** [comment]
I merged the PR to remove *wwwroot* from *.gitignore*. By not checking it in, I think we're setting a bad example for our customers. In the real world, this folder does get checked into source control. 

With this being said, I think it's reasonable to exclude folders like *wwwroot/lib*, since folders/files are copied here during `bower install`. Any opposition to me adding this to *.gitignore*?

**Comment by guardrex** [comment]
@scottaddie You prob saw it ... I just found it when I was looking at the file for something else: https://github.com/aspnet/Docs/blob/master/.gitignore#L210-L211

I still hope you'll reconsider it: If the machine doesn't have Internet access and the fallback is to the *lib* folder and it isn't there ... :boom: :cry:. AFAIK those files would be overridden if replaced at all, so there's no harm AFAICT just leaving the *lib* folder in the samps.

Second, I don't advise tying in unnecessary tech into samples anyway ... a discussion for another day! lol :smile:

File: aspnetcore/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/TestingControllersSample.Tests.csproj [hunk 1]
```diff
@@ -1,54 +1,23 @@
...
-    <RuntimeFrameworkVersion>1.0.4</RuntimeFrameworkVersion>
-    <PackageTargetFallback>$(PackageTargetFallback);dnxcore50;portable-net451+win8</PackageTargetFallback>
-    <GenerateAssemblyTitleAttribute>false</GenerateAssemblyTitleAttribute>
-    <GenerateAssemblyDescriptionAttribute>false</GenerateAssemblyDescriptionAttribute>
-    <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
-    <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
-    <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
-    <GenerateAssemblyCopyrightAttribute>false</GenerateAssemblyCopyrightAttribute>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <OutputType>Exe</OutputType>
```
Comment (scottaddie) [inline aspnetcore/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/TestingControllersSample.Tests.csproj]: Can this node be removed?

Comment (guardrex) [reply aspnetcore/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/TestingControllersSample.Tests.csproj]: Console app

File: aspnetcore/mvc/controllers/testing/sample/TestingControllersSample/src/TestingControllersSample/Views/Home/Index.cshtml [hunk 2]
```diff
@@ -27,12 +27,10 @@
                 Add New Session
             </div>
             <div class="panel-body">
-                @Html.ValidationSummary()
                 <form method="post" asp-controller="Home" asp-action="Index">
                     <fieldset class="form-group">
                         <label for="sessionName">New Session Name</label>
-                        <input type="text" class="form-control" id="sessionName" name="SessionName"
-                               placeholder="New Brainstorming Session Name"/>
+                        <input type="text" class="form-control" id="sessionName" name="SessionName" placeholder="Name" required>
```
Comment (scottaddie) [inline aspnetcore/mvc/controllers/testing/sample/TestingControllersSample/src/TestingControllersSample/Views/Home/Index.cshtml]: Let's be consistent with `input` tags in this file. For example, the button uses a self-closing tag, but this one does not.

Comment (guardrex) [reply aspnetcore/mvc/controllers/testing/sample/TestingControllersSample/src/TestingControllersSample/Views/Home/Index.cshtml]: I missed that one below. I'm aware it's optional; I'll add it back. I just don't close them in my own stuff.

**Comment by guardrex** [comment]
@scottaddie kk ... that one prob is fixed. I'm moving on to fix those last three config samps now ... right after a quick 🚴 ride.

**Pull request merged**